libflux/go/libflux/analyze_test.go

```diff
@@ -56,3 +58,170 @@ func TestAnalyze(t *testing.T) {
 		})
 	}
 }
+
+func TestMergePackages(t *testing.T) {
+	pkg1 := libflux.Parse(`
+package foo
+
+a = 1`)
+	pkg2 := libflux.Parse(`
+package foo
+
+b = 1`)
+	err := libflux.MergePackages(pkg1, pkg2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	gotFB, err := pkg2.MarshalFB()
+	got := ast.DeserializeFromFlatBuffer(gotFB)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := &ast.Package{
+		Package: "foo",
+		Files: []*ast.File{
+			{
+				Name:     "",
+				Metadata: "parser-type=rust",
+				Package: &ast.PackageClause{
+					Name: &ast.Identifier{Name: "foo"},
+				},
+				Body: []ast.Statement{
+					&ast.VariableAssignment{
+						ID:   &ast.Identifier{Name: "b"},
```

Comment:
Hmm, given the way the inputs are arranged, I would have expected the file with `a = 1` to appear first, and `b = 1` to be second.  I wonder if `MergePackages` should just append all the files in the second argument to the files in the first argument?  This way the order of the files would be preserved.